extension/amqp/amqp.go

```diff
@@ -0,0 +1,54 @@
+package amqp
+
+import (
+	"io"
+
+	"github.com/hellofresh/goengine"
+	"github.com/streadway/amqp"
+)
+
+// setup returns a connection and channel to be used for consumption with the Queue setup
+func setup(url, queue string) (*amqp.Connection, *amqp.Channel, error) {
+	conn, err := amqp.Dial(url)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	ch, err := conn.Channel()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if _, err := ch.QueueDeclare(queue, true, false, false, false, nil); err != nil {
+		return nil, nil, err
+	}
+
+	return conn, ch, nil
+}
+
+// DirectQueueConsume returns a Consume func that will connect to the provided AMQP server and create a queue for direct message delivery
+func DirectQueueConsume(amqpDSN, queue string) (Consume, error) {
+	if _, err := amqp.ParseURI(amqpDSN); err != nil {
+		return nil, goengine.InvalidArgumentError("amqpDSN")
+	}
+	if len(queue) == 0 {
+		return nil, goengine.InvalidArgumentError("queue")
+	}
+
+	return func() (io.Closer, <-chan amqp.Delivery, error) {
+		conn, ch, err := setup(amqpDSN, queue)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// Indicate we only want 1 message to be acknowledge at a time.
+		if err := ch.Qos(1, 0, false); err != nil {
+			return nil, nil, err
+		}
+
+		// Exclusive consumer
+		deliveries, err := ch.Consume(queue, "", true, false, false, false, nil)
```

Comment:
The comment above this line suggests that you want to have the `exclusive` argument set, which is here the second `bool` argument, which is `false`. ([streadway/amqp.Channel.Consume](https://godoc.org/github.com/streadway/amqp#Channel.Consume)). So the comment and the code are inconsistent about your intent here.

Additionally, if you intend to have `autoAck` set to true, then the `ch.Qos()` will not do anything, and messages will be delivered as fast as possible without limit, as per documentation for [streadway/amqp.Channel.Qos](https://godoc.org/github.com/streadway/amqp#Channel.Qos):
> With a prefetch count greater than zero, the server will deliver that many messages to consumers before acknowledgments are received. The server ignores this option when consumers are started with noAck because no acknowledgments are expected or sent.